### PR TITLE
chore: upgrade react

### PR DESCRIPTION
**Pull request by parkjh**
- upgrade react to v16.9.0

### CODE REVIEW

File: yarn.lock [hunk 1]
```diff
@@ -17367,7 +17367,17 @@ react-docgen@^3.0.0:
...
-react-dom@16.8.6, react-dom@^16.7.0:
+react-dom@16.9.0:
+  version "16.9.0"
+  resolved "https://registry.yarnpkg.com/react-dom/-/react-dom-16.9.0.tgz#5e65527a5e26f22ae3701131bcccaee9fb0d3962"
+  integrity sha512-YFT2rxO9hM70ewk9jq0y6sQk8cL02xm4+IzYBz75CQGlClQQ1Bxq0nhHF6OtSbit+AIahujJgb/CPRibFkMNJQ==
+  dependencies:
+    loose-envify "^1.1.0"
+    object-assign "^4.1.1"
+    prop-types "^15.6.2"
+    scheduler "^0.15.0"
```
Comment (ValGeorgiev) [inline yarn.lock]: this is good, scheduler is coming <3

File: packages/image/src/safeAreaView/react-native-safe-area.js [hunk 2]
```diff
@@ -154,6 +154,10 @@ class SafeView extends Component {
     _customStatusBarHeight = height;
   };
 
+  static getDerivedStateFromProps() {
+    this._onLayout();
```
Comment (nathggns) [inline packages/image/src/safeAreaView/react-native-safe-area.js]: This isn't going to work right? `_onLayout` is an instance method but `getDerivedStateFromProps` is a class method.

Besides, this is third party code anyway and it has been updated to remove componentWillReceiveProps. Copy it from here https://github.com/react-native-community/react-native-safe-area-view/blob/legacy/index.js

Comment (nathggns) [reply packages/image/src/safeAreaView/react-native-safe-area.js]: This is used in modal images if you want to check it works, btw.

**Comment by parkjh** [comment]
> It's unclear why all the snaps now have extra views in too?

yea me neither... checked with jaye and it's to do with text flow but it is unclear even to him what that view component represent...

File: packages/image/src/safeAreaView/react-native-safe-area.js [hunk 3]
```diff
@@ -1,50 +1,14 @@
-// taken from https://github.com/react-native-community/react-native-safe-area-view
```
Comment (nathggns) [inline packages/image/src/safeAreaView/react-native-safe-area.js]: Can we keep this comment in please.

File: packages/image/src/safeAreaView/react-native-safe-area.js [hunk 4]
```diff
@@ -1,50 +1,14 @@
...
-    };
-
-    render() {
-      return <WrappedComponent {...this.props} {...this.state} />;
-    }
-  }
-
-  return hoistStatics(withOrientationClass, WrappedComponent);
-}
+import withOrientation from "./withOrientation";
```
Comment (nathggns) [inline packages/image/src/safeAreaView/react-native-safe-area.js]: This is probably going to break. You might want to inline this file as previously.

Comment (parkjh) [reply packages/image/src/safeAreaView/react-native-safe-area.js]: copied over

**Comment by nathggns** [comment]
Why did you assign this to me?

**Comment by parkjh** [comment]
> Why did you assign this to me?

to rereview?

**Comment by nathggns** [comment]
@parkjh you need to understand what is causing the extra views before this can be approved.

**Comment by parkjh** [comment]
> @parkjh you need to understand what is causing the extra views before this can be approved.

@jmars ^^^

**Comment by jmars** [comment]
It's the height:0 that makes me suspicious it's related to text, but why would a reat upgrade cause this? Seems strange, looking into it now

File: packages/article-skeleton/__tests__/android/__snapshots__/article-with-style.android.test.js.snap [hunk 5]
```diff
@@ -669,32 +669,7 @@ exports[`2. an article with interactives 1`] = `
 </View>
 `;
 
-exports[`3. an article with no content 1`] = `
```
Comment (parkjh) [inline packages/article-skeleton/__tests__/android/__snapshots__/article-with-style.android.test.js.snap]: @jmars I know that this is intentional but is this the correct behaviour?

**Pull request merged**